helpers.go

```diff
@@ -0,0 +1,160 @@
+package option
+
+import (
+	"github.com/vmihailenco/msgpack/v5"
+	"github.com/vmihailenco/msgpack/v5/msgpcode"
+)
+
+func checkNumber(code byte) bool {
+	switch {
+	case msgpcode.IsFixedNum(code):
+		return true
+	case code == msgpcode.Int8 || code == msgpcode.Int16 || code == msgpcode.Int32 || code == msgpcode.Int64:
+		return true
+	case code == msgpcode.Uint8 || code == msgpcode.Uint16 || code == msgpcode.Uint32 || code == msgpcode.Uint64:
+		return true
+	default:
+		return false
+	}
+}
+
+func decodeInt(decoder *msgpack.Decoder) (int, error) {
+	return decoder.DecodeInt()
+}
+
+func decodeInt8(decoder *msgpack.Decoder) (int8, error) {
+	return decoder.DecodeInt8()
+}
+
+func decodeInt16(decoder *msgpack.Decoder) (int16, error) {
+	return decoder.DecodeInt16()
+}
```

Comment:
Please, add a comment: why do we need such helper functions? It is not obvious. Probably the comment could be placed into the header of the file.